osg-comanage-project-usermap.py

```diff
@@ -146,6 +149,35 @@ def get_osguser_groups(filter_group_name=None):
              for pid, gids in pid_gids.items() }
 
 
+def parse_inputfile(inputfile):
+    user_groupmap = dict()
+    with open(inputfile) as file:
+        for line in file:
+            split_line = line.split()
+            if split_line[0] == "*" and len(split_line) >= 3:
+                user_groupmap[split_line[1]] = split_line[2].split(",")
+    return user_groupmap
+
+
+def merge_maps(maps):
+    while len(maps) > 1:
+        merge_target = maps[0]
+        merge_material = maps.pop(1)
+        for key in merge_material.keys():
+            if key in merge_target:
+                merge_target[key].extend(merge_material[key])
+                merge_target[key] = list(set(merge_target[key]))
```

Comment:
I think the appropriate data structure is just a dict with users as the keys and projects stored as a `set`. You get the deduplication for free and I can't think of anything list-specific that you need.